frontend/src/components/Layout/Header/styled.ts

```diff
@@ -13,90 +28,217 @@ export const LayoutHeader = styled(Grid)({
   },
   [theme.breakpoints.up('md')]: {
-    margin: '28px 30px 28px 30px',
+    width: '105px',
+    margin: '19px 30px 16px 32px',
+  },
+  '@media(min-width: 1050px)': {
+    margin: '19px 48px 16px 32px',
   },
   [theme.breakpoints.up('xl')]: {
+    width: '130px',
     margin: '28px 163px 28px 40px',
   },
 });
 export const BrandLogo = styled('img')({
-  width: '130px',
+  [theme.breakpoints.up('xs')]: {
+    width: '79px',
+  },
+  [theme.breakpoints.up('md')]: {
+    width: '105px',
+  },
+  [theme.breakpoints.up('xl')]: {
+    width: '130px',
+  },
 });
 export const HeaderContent = styled('div')({
   flexGrow: '2',
   flexShrink: '2',
-
   display: 'flex',
   justifyContent: 'flex-end',
   height: HEADER_HEIGHT,
   textAlign: 'right',
+  position: 'relative',
+  [theme.breakpoints.down('md')]: {
+    width: 'calc(100% - 101px)',
+  },
 });
 export const SpaceHolder = styled('div')({
-  flexGrow: '1',
-  flexShrink: '3',
-  maxWidth: '1000px',
+  flexGrow: '4',
+  flexShrink: '2',
   height: HEADER_HEIGHT,
+  [theme.breakpoints.up('xs')]: {
+    display: 'none',
+  },
+  [theme.breakpoints.up('lg')]: {
+    display: 'block',
+    maxWidth: '1000px',
+  },
 });
-export const Search = styled(Input)({
-  flexGrow: '2',
-  flexShrink: '3',
-  height: '50px',
+export const Search = styled(Input)<CoursesPageProps>(({ pagename, menuopen }) => ({
+  flexGrow: '0',
+  flexShrink: '1',
   borderRadius: '3px',
-  margin: '15px 20px 15px 0px',
   padding: '10px',
-  fontSize: '24px',
   backgroundColor: 'rgba(118, 118, 128, 0.12)',
   color: '#3c3c43',
   [theme.breakpoints.up('xs')]: {
     display: 'none!important',
+    position: 'relative',
+    ...(pagename === PAGES_TO_SEARCH.myCourses &&
+      !menuopen && {
+        flexShrink: '0',
+        display: 'block',
+        position: 'absolute',
+        width: '260px',
+        height: '30px',
+        top: '60px',
+        right: 'calc(100vw - 310px)',
+      }),
+    ...(pagename === PAGES_TO_SEARCH.coursesList &&
+      !menuopen && {
+        flexShrink: '0',
+        display: 'block',
+        position: 'absolute',
+        width: '260px',
+        height: '30px',
+        top: '60px',
+        right: 'calc(100vw - 310px)',
+      }),
+  },
+  [theme.breakpoints.up('md')]: {
+    position: 'relative',
+    top: '0px',
+    right: '0px',
+    display: 'inline-flex!important',
+    width: '200px!important',
+    height: '44px',
+    fontSize: '16px',
+    margin: '8px 20px 8px 0px',
+  },
+  '@media(min-width: 900px)': {
+    width: '300px!important',
   },
   [theme.breakpoints.up('lg')]: {
+    position: 'relative',
     display: 'inline-flex!important',
-    maxWidth: '463px!important',
+    width: '463px!important',
+    height: '44px',
+    fontSize: '16px',
+    margin: '8px 20px 8px 0px',
   },
   [theme.breakpoints.up('xl')]: {
-    maxWidth: '730px!important',
+    height: '50px',
+    fontSize: '24px',
+    margin: '15px 20px 15px 0px',
   },
-});
+  '@media(min-width: 1530px)': {
```

Comment:
We have problem with "jump down" search menu on this and nearby resolutions(1528-1555px). Maybe will fix `width` of search block?

Reply:
fixed